builder/fscache/fscache.go

```diff
@@ -607,3 +616,40 @@ func (s sortableCacheSources) Less(i, j int) bool {
 func (s sortableCacheSources) Swap(i, j int) {
 	s[i], s[j] = s[j], s[i]
 }
+
+func newTarsumHash(stat *fsutil.Stat) (hash.Hash, error) {
+	fi := &fsutil.StatInfo{stat}
+	p := stat.Path
+	if fi.IsDir() {
+		p += string(os.PathSeparator)
```

Comment:
Should this try to trim any trailing os.PathSeprator first in case it already has the suffix?

Reply:
These paths are already cleaned https://github.com/moby/moby/pull/34943/files/d4729749023fde0c57cded0c8159dc85cd7ee448#diff-78afa8e80f1f2221c7bf7bc880b54f70R33